packages/dartboard/src/Dartboard.js

```diff
@@ -81,127 +212,163 @@ const Dartboard = ({
         secondInput
       });
     } catch (err) {
-      return events.error(response?.text() || {
-        message: err.message,
-        status: 400
-      });
+      return props.events.error(
+        response?.text() || {
+          message: err.message,
+          status: 400
+        }
+      );
     }
 
     const location = await extractResponse(response);
 
     if (location) {
-      return events.success(location);
+      return props.events.success(location);
     }
-  };
+  }, [firstInput, secondInput, validate, props.events, get, extractResponse]);
+
+  const handleKeyPress = useCallback((event) => {
+    if (event.key !== 'Enter') {
+      return;
+    }
+
+    find();
+  }, [find]);
 
-  const get = options => {
+  const get = useCallback((options) => {
     const url = `${baseUrl}/${options.firstInput}/${options.secondInput}?`;
 
-    const query = {
-      apiKey: apiKey,
-      spatialReference: wkid
+    let query = {
+      apiKey: props.apiKey,
+      spatialReference: props.wkid,
+      format: props.format,
+      callback: props.callback
     };
 
+    if (props.type === ADDRESS_TYPE) {
+      query = { ...props.address, ...query };
+    } else {
+      query = { ...props.milepost, ...query };
+    }
+
     const querystring = toQueryString(query);
 
     return fetch(url + querystring, {
       method: 'GET',
       mode: 'cors'
     });
-  };
+  }, [props.apiKey, props.wkid, props.address, props.milepost, props.type, props.format, props.callback, baseUrl]);
 
-  const extractResponse = async response => {
+  const extractResponse = useCallback(async (response) => {
     if (!response.ok) {
       setFound(false);
 
-      return events.error(await response.json());
+      return props.events.error(await response.json());
     }
 
     let result = await response.json();
 
     if (result.status !== 200) {
       setFound(false);
 
-      return events.error(response);
+      return props.events.error(response);
     }
 
     result = result.result;
 
+    if (props.format?.toLowerCase() === 'geojson') {
+      return result;
+    }
+
+    setFound(true);
+
     const point = {
       type: 'point',
       x: result.location.x,
       y: result.location.y,
       spatialReference: {
-        wkid: wkid
+        wkid: props.wkid
       }
     };
 
     return outputTransform(result, point);
-  };
+  }, [outputTransform, props.wkid, props.format, props.events]);
 
-  const validate = () => {
-    const firstValidity = firstInput.trim().length > 0;
-    const secondValidity = secondInput.trim().length > 0;
+  const validate = useCallback(() => {
+    const firstValidity = firstInput?.trim().length > 0;
+    const secondValidity = secondInput?.trim().length > 0;
 
     setFirstIsValid(firstValidity);
     setSecondIsValid(secondValidity);
 
     // reset not found message
-    setFound(true);
+    setFound(undefined);
```

Comment:
**thought**: I wonder if this should be `null` since it's a deliberate setting of the value rather than something that's not defined.

```suggestion
    setFound(null);
```